ewatercycle/models/lisflood.py

```diff
@@ -0,0 +1,398 @@
+    def run_lisvap(self, forcing: PathLike) -> Tuple[int, bytes, bytes]:
+        """Run lisvap to generate evaporation input files
+
+        Args:
+            forcing: Path to re-indexed forcings
+
+        Returns:
+            Tuple with exit code, stdout and stderr
+        """
+        singularity_image = CFG['lisflood.singularity_image']
+        docker_image = CFG['lisflood.docker_image']
+        self._check_forcing(forcing)
+        lisvap_file = self._create_lisvap_config()
+
+        mount_points = {
+            f'{self.parameterset.root}': '/data/lisflood_input',
+            f'{self.parameterset.mask}': '/data/model_mask.nc',
+            f'{self.forcing_dir}': '/data/forcing',
+            f'{self.work_dir}': '/output',
+        }
+
+        if CFG['container_engine'].lower() == 'singularity':
+            args = [
+                "singularity",
+                "exec",
+            ]
+            args += ["--bind", ','.join([hp + ':' + ip for hp, ip in mount_points.items()])]
+            args.append(singularity_image)
+
+        elif CFG['container_engine'].lower() == 'docker':
+            args = [
+                "docker",
+                "run -ti",
+            ]
+            args += ["--volume", ','.join([hp + ':' + ip for hp, ip in mount_points.items()])]
+            args.append(docker_image)
+
+        else:
+            raise ValueError(
+                f"Unknown container technology in CFG: {CFG['container_engine']}"
+            )
+
+        lisvap_filename = Path(lisvap_file).name
+        args += ['python3', '/opt/Lisvap/src/lisvap1.py', f"/output/{lisvap_filename}"]
+        container = subprocess.Popen(args, preexec_fn=os.setsid, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        exit_code = container.wait()
+        stdout, stderr = container.communicate()
+        return exit_code, stdout, stderr
+
+    def get_value_as_xarray(self, name: str) -> xr.DataArray:
+        """Return the value as xarray object."""
+        # Get time information
+        time_units = self.bmi.get_time_units()
+        grid = self.bmi.get_var_grid(name)
+        shape = self.bmi.get_grid_shape(grid)
+
+        # Extract the data and store it in an xarray DataArray
+        da = xr.DataArray(
+            data=np.reshape(self.bmi.get_value(name), shape),
+            coords={
+                "longitude": self.bmi.get_grid_y(grid),
+                "latitude": self.bmi.get_grid_x(grid),
+                "time": num2date(self.bmi.get_current_time(), time_units)
+            },
+            dims=["latitude", "longitude"],
+            name=name,
+            attrs={"units": self.bmi.get_var_units(name)},
+        )
+
+        return da
+
+    @property
+    def parameters(self) -> Iterable[Tuple[str, Any]]:
+        """List the parameters for this model."""
+        if self.forcing_dir and self.work_dir:
+            return {
+                'forcing_dir': self.forcing_dir,
+                'work_dir': self.work_dir,
```

Comment:
I have a bit of an internal conflict about the naming of `work_dir` (which is what grpc4bmi uses) and `cfg_dir` (which is what PyMT uses). I'm inclined to go with the latter for everything that the user interacts with, to stay in line with PyMT.

Reply:
I agree. In [PyMT usage](https://pymt.readthedocs.io/en/latest/usage.html), the variable is named as `cfg_dir`. However, it is referred to **run directory** in the text. Let's make an issue for this and discuss it in a design session.